llama_index/llms/localai.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+
+from llama_index.llms.openai import OpenAI
+from llama_index.bridge.pydantic import Field
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW
+
+
+class LocalAI(OpenAI):
+    """
+    LocalAI is a free, open source, and self-hosted OpenAI alternative.
+
+    Docs: https://localai.io/
+    Source: https://github.com/go-skynet/LocalAI
+    """
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "LocalAI"
+
+    context_window: int = Field(
+        default=DEFAULT_CONTEXT_WINDOW,
+        description="The maximum number of context tokens for the model.",
+    )
+
+    def __init__(self, context_window: int = DEFAULT_CONTEXT_WINDOW, **openai_kwargs):
+        super().__init__(**openai_kwargs)
+        self.context_window = context_window  # Set in pydantic
+
+    def _get_context_window(self) -> int:
+        return self.context_window
+
+    def _get_max_token_for_prompt(self, prompt: str) -> Optional[int]:
+        return None
```

Comment:
@logan-markewich can you help me out with this one?  For a generic LocalAI model like [this one](https://huggingface.co/TheBloke/Llama-2-13B-Ensemble-v5-GGUF/blob/main/llama-2-13b-ensemble-v5.Q4_K_M.gguf), I am not sure what to set `max_tokens`

I think leaving it as `null` is fine, as that's what I have always done for LocalAI

Reply:
One observation is it's not really `null`, it can just be excluded from the POST's data.  Going to resolve this comment